Objects/dictobject.c

```diff
@@ -5553,11 +5594,20 @@ _PyObject_StoreInstanceAttribute(PyObject *obj, PyDictValues *values,
     assert(values != NULL);
     assert(Py_TYPE(obj)->tp_flags & Py_TPFLAGS_MANAGED_DICT);
 
-    if(!Py_CHECKWRITE(obj)){
+    if (!Py_CHECKWRITE(obj)){
         PyErr_WriteToImmutable(obj);
         return -1;
     }
 
+    //TODO: PYRONA: The addition of the key is complex here.
+    //  The keys PyDictKeysObject, might already have the key. Note that
+    //  the keys PyDictKeysObject is not a PyObject. So it is unclear where
+    //  this edge is created.
+    //  The keys is coming from ht_cached_keys on the type object.
```

Comment:
```suggestion
    //  The key is coming from ht_cached_keys on the type object.
```

Reply:
I actually meant the plural, as it is the "keys object/structure".  But I can see why it is confusing and the addition of object/structure would make it clearer.